pkg/blockatlas/marketdata.go

```diff
@@ -0,0 +1,58 @@
+package blockatlas
+
+import (
+	"math/big"
+	"time"
+)
+
+const (
+	TypeCoin  CoinType = "coin"
+	TypeToken CoinType = "token"
+)
+
+type CoinType string
+
+type TickerResponse struct {
+	Currency string  `json:"currency"`
+	Result   Tickers `json:"result"`
+}
+
+type Ticker struct {
+	Coin       uint        `json:"coin"`
+	CoinName   string      `json:"coin_name,omitempty"`
+	TokenId    string      `json:"token_id,omitempty"`
+	CoinType   CoinType    `json:"type,omitempty"`
+	Price      TickerPrice `json:"price,omitempty"`
+	LastUpdate time.Time   `json:"last_update,omitempty"`
+	Error      string      `json:"error,omitempty"`
+}
+
+type TickerPrice struct {
+	Value     *big.Float `json:"value,omitempty"`
```

Comment:
any reason to use big float here? decimal values usually does not require big number, not for this use case

Reply:
alway we need to use a number with a lot of decimals number and do some calc with them, it's better to use big numbers. It's more safety and you have less round and approximate values